qiskit/quantum_info/operators/unitary.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name,assignment-from-no-return
+
+"""
+Tools for working with Unitary Operators.
+
+A simple unitary class and some tools.
+"""
+import copy
+import numpy
+import sympy
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.gate import Gate
+from qiskit.dagcircuit import DAGCircuit
+
+
+class Unitary(Gate):
+    """Class for representing unitary operators"""
+
+    def __init__(self, matrix, *qargs, label=None, validate=True, rtol=1e-5, atol=1e-8):
+        """
+        Create unitary.
+
+        Args:
+            matrix (numpy.ndarray or list(list)): unitary matrix representation.
+            qargs (list(tuple(QuantumCircuit, int))): qubits to apply operation to.
+            label (str): identifier hint for backend
+            validate (bool): whether to validate unitarity of matrix when supplied
+                either here or with Unitary.matrix.
+            rtol (float): relative tolerance (see numpy.allclose).
+            atol (float): absolute tolerance (see numpy.allclose).
+        """
+        self._matrix = None
+        self._label = None
+        self._validate = validate
+        self._rtol = rtol
+        self._atol = atol
+        self._n_qubits = None
+        self._dimension = None
+        # set matrix (depends on previous attributes)
+        self.matrix = matrix
+        self.label = label
+        self._decompostion = []  # storage (needed?)
+        super().__init__('unitary', [sympy.Matrix(matrix)], list(qargs))
+
+    @property
+    def dimension(self):
+        """dimension of matrix
+
+        Returns:
+            int: dimension of matrix
+        """
+        return self._dimension
+
+    def tensor(self, other, front=False):
```

Comment:
@chriseclectic and I were talking about this. Seems like there is a use case for having a `expand()` method that just does the opposite of `tensor()` in terms of ordering:
```
A.tensor(B)   # literally does a mathematical A ⊗ B.
A.expand(B)  # this is more like our QuantumCircuit(A, B) convention. it expands the subspace of A by adding B as a more significant qubit. mathematically: B ⊗ A

Reply:
I think Chris will do this in the quantum channels package, but can you also do it here for the unitary.

Reply:
done